flink-table-store-spark/src/main/java/org/apache/flink/table/store/spark/SparkCatalog.java

```diff
@@ -291,8 +291,19 @@ private UpdateSchema toUpdateSchema(
                         : Arrays.stream(pkAsString.split(","))
                                 .map(String::trim)
                                 .collect(Collectors.toList());
+        RowType rowType = (RowType) toFlinkType(schema);
+        List<String> fields = rowType.getFieldNames();
+        if (!primaryKeys.isEmpty()) {
```

Comment:
We can add this check in UpdateSchema.
- Primary / partition keys should in fields.
- Pk not null. (Or convert pk fields to not null)